web/server/api/endpoints/lineage.py

```diff
@@ -55,9 +75,43 @@ async def column_lineage(
 
     graph = {}
     table = model
+    cache_column_names: t.Dict[str, str] = {}
+
     for i, node in enumerate(node.walk()):
         if i > 0:
-            table = _get_table(node)
+            table = _get_table(node) or model
             column = exp.to_column(node.name).name
-        graph[table] = {column: _process_downstream(node.downstream)}
+        if column in cache_column_names:
+            column = cache_column_names[column]
+        graph[table] = {
+            column: LineageColumn(
+                source=_get_node_source(node=node, dialect=context.models[table].dialect),
+                models=_process_downstream(
+                    node.downstream,
+                    column,
+                    cache_column_names,
+                ),
+            )
+        }
     return graph
+
+
+@router.get("/{model_name:str}")
+async def model_lineage(
```

Comment:
What does this do? `models` only looks one level upstream and one level downstream. By the way, model lineage is already implemented: ` return context.dag.lineage(model_name).graph`. If you change the return type to `t.Dict[str, t.Set[str]]`, the framework can handle serializing the sets.